proxy/src/transport/tls/config.rs

```diff
@@ -66,6 +75,33 @@ pub struct ServerConfig(pub(super) Arc<rustls::ServerConfig>);
 pub type ClientConfigWatch = Watch<Option<ClientConfig>>;
 pub type ServerConfigWatch = Watch<Option<ServerConfig>>;
 
+/// The configuration in effect for a client (`ClientConfig`) or server
+/// (`ServerConfig`) TLS connection.
+#[derive(Clone, Debug)]
+pub struct ConnectionConfig<C> where C: Clone + std::fmt::Debug {
```

Comment:
Another potential alternative to adding a generic `Conditional` type could be just changing this to an enum 
```rust
pub enum ConnectionConfig<C>
   Some { identity: Identity, config: C },
   None(ReasonForNoTls)
}
```
or similar...